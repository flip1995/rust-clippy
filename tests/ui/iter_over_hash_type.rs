//@aux-build:proc_macros.rs

#![warn(clippy::iter_over_hash_type)]
use std::collections::{HashMap, HashSet};

extern crate proc_macros;

fn main() {
    let mut hash_set = HashSet::<i32>::new();
    let mut hash_map = HashMap::<i32, i32>::new();
    let vec = Vec::<i32>::new();

    for x in &hash_set {
        let _ = x;
    }
    for x in hash_set.iter() {
        let _ = x;
    }
    for x in hash_set.clone() {
        let _ = x;
    }
    for x in hash_set.drain() {
        let _ = x;
    }
    for (x, y) in &hash_map {
        let _ = (x, y);
    }
    for x in hash_map.keys() {
        let _ = x;
    }
    for x in hash_map.values() {
        let _ = x;
    }
    for x in hash_map.values_mut() {
        *x += 1;
    }
    for x in hash_map.iter() {
        let _ = x;
    }
    for x in hash_map.clone() {
        let _ = x;
    }
    for x in hash_map.drain() {
        let _ = x;
    }

    // shouldnt fire
    for x in &vec {
        let _ = x;
    }
    for x in vec {
        let _ = x;
    }

    // should not lint, this comes from an external crate
    proc_macros::external! {
      for _ in HashMap::<i32, i32>::new() {}
    }
}
