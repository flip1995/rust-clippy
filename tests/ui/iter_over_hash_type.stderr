error: iteration over unordered hash-based type
  --> $DIR/iter_over_hash_type.rs:13:5
   |
LL | /     for x in &hash_set {
LL | |         let _ = x;
LL | |     }
   | |_____^
   |
   = note: `-D clippy::iter-over-hash-type` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::iter_over_hash_type)]`

error: iteration over unordered hash-based type
  --> $DIR/iter_over_hash_type.rs:16:5
   |
LL | /     for x in hash_set.iter() {
LL | |         let _ = x;
LL | |     }
   | |_____^

error: iteration over unordered hash-based type
  --> $DIR/iter_over_hash_type.rs:19:5
   |
LL | /     for x in hash_set.clone() {
LL | |         let _ = x;
LL | |     }
   | |_____^

error: iteration over unordered hash-based type
  --> $DIR/iter_over_hash_type.rs:22:5
   |
LL | /     for x in hash_set.drain() {
LL | |         let _ = x;
LL | |     }
   | |_____^

error: iteration over unordered hash-based type
  --> $DIR/iter_over_hash_type.rs:25:5
   |
LL | /     for (x, y) in &hash_map {
LL | |         let _ = (x, y);
LL | |     }
   | |_____^

error: iteration over unordered hash-based type
  --> $DIR/iter_over_hash_type.rs:28:5
   |
LL | /     for x in hash_map.keys() {
LL | |         let _ = x;
LL | |     }
   | |_____^

error: iteration over unordered hash-based type
  --> $DIR/iter_over_hash_type.rs:31:5
   |
LL | /     for x in hash_map.values() {
LL | |         let _ = x;
LL | |     }
   | |_____^

error: iteration over unordered hash-based type
  --> $DIR/iter_over_hash_type.rs:34:5
   |
LL | /     for x in hash_map.values_mut() {
LL | |         *x += 1;
LL | |     }
   | |_____^

error: iteration over unordered hash-based type
  --> $DIR/iter_over_hash_type.rs:37:5
   |
LL | /     for x in hash_map.iter() {
LL | |         let _ = x;
LL | |     }
   | |_____^

error: iteration over unordered hash-based type
  --> $DIR/iter_over_hash_type.rs:40:5
   |
LL | /     for x in hash_map.clone() {
LL | |         let _ = x;
LL | |     }
   | |_____^

error: iteration over unordered hash-based type
  --> $DIR/iter_over_hash_type.rs:43:5
   |
LL | /     for x in hash_map.drain() {
LL | |         let _ = x;
LL | |     }
   | |_____^

error: aborting due to 11 previous errors

